torch/distributed/distributed_c10d.py

```diff
@@ -2369,7 +2369,7 @@ def gather_object(obj, object_gather_list=None, dst=0, group=None):
             should be correctly sized as the size of the group for this
             collective and will contain the output. Must be ``None`` on non-dst
             ranks. (default is ``None``)
-        dst (int, optional): Destination rank. (default is 0)
+        dst (int, optional): Destination rank on global process group (regardless of 'group' argument). (default is 0)
```

Comment:
Even if the document is updated, the behavior is very counterintuitive. Should we change to use the group's rank?

Reply:
I prefer the changed behavior.  But i do not know if it is worth the disruption of making a breaking change. worth a discussion

Reply:
if we want to change it, what we might have to do is first add a 'group_rank' arg and add logic to ensure only one of group_rank or rank is passed.  Then deprecate rank.  then finally remove rank.  Or how else?